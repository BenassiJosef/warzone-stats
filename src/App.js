import "./App.css";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Search from "./pages/Search/Search";
import { Box, Grommet } from "grommet";
import { FONT_FAMILY, TEXT_FONT_SIZE } from "./constants";

const AppBar = (props) => (
  <Box
    tag="header"
    direction="row"
    align="center"
    justify="between"
    background="#f5f5f5"
    pad={{ left: "medium", right: "small", vertical: "small" }}
    elevation="large"
    style={{ zIndex: "1" }}
    {...props}
  />
);

const theme = {
  global: {
    font: {
      family: FONT_FAMILY,
      size: TEXT_FONT_SIZE,
    },
  },
};

function App() {
  return (
    <Grommet theme={theme} themeMode="dark">
      <AppBar>Warzone Stats</AppBar>
      <Router>
        <Switch>
          <Route exact path="/" component={Search} />
        </Switch>
      </Router>
    </Grommet>
  );
}

export default App;
