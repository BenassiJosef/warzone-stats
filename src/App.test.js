import React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import App from "./App";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

Enzyme.configure({ adapter: new EnzymeAdapter() });

test("renders without crashing", () => {
  const store = mockStore({});
  const wrapper = shallow(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(wrapper).toBeTruthy();
});

test("snapshot renders", () => {
  const store = mockStore({ searchReducer: { searchResponse: {} } });
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
