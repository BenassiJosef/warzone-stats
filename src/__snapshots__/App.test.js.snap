// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="App"
>
  <div
    className="sc-dlfnbm HlyKS"
  >
    <div
      style={
        Object {
          "marginBottom": "8px",
        }
      }
    >
      <label
        className="sc-gsTCUz jFhTAk"
        htmlFor="player"
      >
        Activision Player ID Examlpe: (josefbenassi%237491959)
      </label>
    </div>
    <input
      aria-label="Activision Player ID Examlpe: (josefbenassi%237491959)-aria"
      aria-required="true"
      className="sc-bdfBwQ kmKmLh"
      id="player"
      label="Activision Player ID Examlpe: (josefbenassi%237491959)"
      onChange={[Function]}
      readOnly={false}
      required={true}
    />
  </div>
  <button
    className="sc-hKgILt AWien"
    data-test="primary-button"
    disabled={false}
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
